Handle move blocking

sir-marc opened the pull request.

sir-marc committed:
Extend gameObject interface with moveBlocking prop

sir-marc committed:
Check moveBlocking prop in moveIntent

michaelschufi committed:
AB#92 Backend Persistence Handling (#47)

* letting the client request its old session

* fixed based on feedback

* Extract ConnectionId to darwin-types

* fix tests

* added tests

* typing for matchUpdates

* cleaned up tests again

* renamed UserContextId to UserId and merged it with connectionId

* refactored server too

* deleted unnecessary type

Co-authored-by: Simon Stucki <simon_stucki@hotmail.com>

michaelschufi committed:
AB#92 Backend Persistence Handling (#47)

* letting the client request its old session

* fixed based on feedback

* Extract ConnectionId to darwin-types

* fix tests

* added tests

* typing for matchUpdates

* cleaned up tests again

* renamed UserContextId to UserId and merged it with connectionId

* refactored server too

* deleted unnecessary type

Co-authored-by: Simon Stucki <simon_stucki@hotmail.com>

StuckiSimon committed:
Implement Persistence Frontend Handling (#52)

mailaenderli committed:
#AB71 move to food (#48)

* implement food selector

* add food-selector tests

* implement userUnit-selector

* unify state-selector imports

* add userUnit-selector tests

* update help text

* Use pre html tag for CodeBlock.tsx

Preserves whitespace in code snippets.

* Improve HelpText

* improve coding style according to PR review

* Refactor gameObjects.ts helper

* move tests to gameObjects.test.ts

* improved tests based on PR feedback

Co-authored-by: Simon Stucki <simon_stucki@hotmail.com>

StuckiSimon commented on darwin-types/game-objects/GameObject.ts:
```diff
@@ -6,6 +6,7 @@ export interface GameObject {
   id: ObjectId;
   type: string;
   position: Position;
+  moveBlocking: boolean;
```
Is there any purpose permitting `moveBlocking` to be set on every instance?

**If not:**
We now have two implementations (one in test and one in app)
I'd suggest to create for each type **one** builder which gets used all over the application (also in tests :v:)

StuckiSimon committed:
Use devicePixelRatio for canvas resolution (#53)

Makes canvas shapes appear sharper.

StuckiSimon committed:
Use devicePixelRatio for canvas resolution (#53)

Makes canvas shapes appear sharper.

sir-marc committed:
AB#83 food frontend rendering (#51)

* Add render layer order

* Wrap PixiGraphics component in custom component

* Add fill property to geometric graphics

* Refactor and redesign canvas

* Add triangle component

* Make Unit representation a bit fancier

* Add ownership differentiating styles

sir-marc committed:
Extract gameObject creation into helper function

StuckiSimon committed:
Tune Canvas Element Styling (#54)

Change coloring and visualization of game objects.
Use inner shapes such as parallelogram.

sir-marc committed:
Merge branch 'master' into handle-move-blocking

sir-marc committed:
Fix linting errors

sir-marc committed:
Fix unit creation helper

StuckiSimon reviewed (approved):
Looks good now 👍
